base/reducedim.jl

```diff
@@ -57,24 +57,33 @@ function reduced_indices0(inds::Indices{N}, region) where N
     tuple(rinds...)::typeof(inds)
 end
 
-
 ###### Generic reduction functions #####
 
+## extensions to the standard functions for BigInt, String and others
+
+zero2(t::Type) =  zero(t)
+zero2(::Type{T}) where T<:Number = zero(T)
+zero2(::Type{T}) where T<:AbstractString = ""
```

Comment:
The way is not to rely on zero behavior at all, but to use the first slice as the start value.